pymeasure/instruments/thermotron/thermotron3800.py

```diff
@@ -0,0 +1,129 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2021 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import strict_discrete_set, strict_range
+
+class Thermotron3800(Instrument):
+    """ Represents the Thermotron 3800 Oven.
+    For now, this driver only supports using Control Channel 1.
+    """
+
+    def __init__(self, resourceName, **kwargs):
+        super().__init__(
+            resourceName,
+            "Thermotron 3800",
+            **kwargs
+        )
+
+    id = Instrument.measurement(
+        "IDEN?", """ Reads the instrument identification 
+        
+        :return: String
+        """
+    )
+
+    temperature = Instrument.measurement(
+        "PVAR1?", """ Reads the current temperature of the oven 
+        via built in thermocouple
+        
+        :return: float
+        """
+    )
+
+    mode = Instrument.measurement(
+        "MODE??", """ Gets the operating mode of the oven.
+        
+        :return: Tuple(String, int)
+        """,
+        get_process=lambda mode: Thermotron3800.__translate_mode(mode)
```

Comment:
`get_process=Thermotron3800.__translate_mode` should be sufficient I think (instead of the one-arg pass-through lambda).

Reply:
Thanks for the suggestion! I tried removing the one-arg pass-through lambda, but I got an error (Thermotron3800 not defined) when I ran it. I tried playing around with different schemes, but I couldn't figure out how to get it to work without the lambda function.

Reply:
Ah, yeah. For that to work `__translate_mode` would have to be a standalone function, not a class method, because at import time at this line, `Thermotron3800` does not yet exist (because the class is not fully defined, yet). Sorry, I forgot about that detail.

Reply:
Got it, thanks for explaining! I wasn't sure why, so that was helpful :)